arbitrum-docs/run-arbitrum-node/run-nitro-dev-node.mdx

```diff
@@ -0,0 +1,168 @@
+
+To set the chain owner, you can call the `becomeChainOwner()` function with your preffered private key, we'll use the dev account as an example:
+
+```bash
+cast send 0x00000000000000000000000000000000000000FF "becomeChainOwner()" --private-key 0xb6b15c8cb491557369f3c7d2c287b053eb229daa9c22138887752191c9520659 --rpc-url http://127.0.0.1:8547
+```
+
+- `0x00000000000000000000000000000000000000FF`: This is the address of the ArbDebug precompile on Arbitrum.
+- `becomeChainOwner()`: This is the function call that transfers ownership of the chain to the caller.
+- `--private-key`: Use the private key of the account that should become the chain owner.
+- `--rpc-url http://127.0.0.1:8547`: Ensure that this points to your local Nitro dev node running in `--dev` mode.
+
+Once you're the chain owner, you'll be able to access various Arbitrum-specific functions, such as:
+
+- Managing the Sequencer
+- Modifying chain parameters
+- Accessing debugging features
+
+This functionality is especially useful for development and testing purposes when using Nitro in `--dev` mode.
+
+## Running Nitro dev node for Stylus testing
+
+If your goal is to test Stylus contracts on Nitro in `--dev` mode, you can set up the environment in the same way, but with the added step of deploying the `Stylus Cache Manager`, which is necessary for Stylus-specific functionality.
+
+To automate this process, you can use a bash script that both runs the Nitro dev node and deploys the `Stylus Cache Manager` contract automatically. Here's how you can do it:
+
+### Step 1: Create a Bash script to automate the process
+
+Create a new file called `run_nitro_with_cache_manager.sh` and add the following content:
+
+```bash
+#!/bin/bash
+
+# Start Nitro Dev Node in the background
+echo "Starting Nitro dev node..."
+docker run --rm --name nitro-dev -p 8547:8547 offchainlabs/nitro-node:v3.2.1-d81324d --dev --http.addr 0.0.0.0 &
+
+# Wait for the node to initialize
+echo "Waiting for the Nitro node to initialize..."
+sleep 15  # Adjust the sleep time if necessary based on node startup time
+
+# Check if node is running
+curl_output=$(curl -s -X POST -H "Content-Type: application/json" \
+  --data '{"jsonrpc":"2.0","method":"net_version","params":[],"id":1}' \
+  http://127.0.0.1:8547)
+
+if [[ "$curl_output" == *"result"* ]]; then
+  echo "Nitro node is running!"
+else
+  echo "Failed to start Nitro node."
+  exit 1
+fi
+
+echo "Deploying Cache Manager contract..."
```

Comment:
This deploys the cache manager, but then it needs to be registered via the owner calling https://github.com/OffchainLabs/nitro-contracts/blob/fbbcef09c95f69decabaced3da683f987902f3e2/src/precompiles/ArbOwner.sol#L130